api/modules/publishedProjects/controller.js

```diff
@@ -0,0 +1,55 @@
+var Boom = require('boom');
+
+var Errors = require('../../classes/errors');
+var BaseController = require('../../classes/base_controller');
+var Model = require('./model');
+var Project = require('../projects/model');
+var File = require('../files/model');
+var controller = new BaseController(Model);
+
+function copyFile(file, projectId) {
+  return File.forge({
+    path: file.get('path'),
+    project_id: projectId,
+    buffer: file.get('buffer')
+  }).save()
+  .then(function(projectFile) {
+    if(!projectFile) { throw Boom.notFound(); }
+  });
+}
+
+controller.clone = function(req, reply) {
+  var user = req.pre.user;
+  var publishedProject = req.pre.records.models[0];
+  var now = (new Date()).toISOString();
+  var project;
+
+  var result = Project.forge({
+    title: publishedProject.get('title'),
+    user_id: user.get('id'),
+    tags: publishedProject.get('tags'),
+    description: publishedProject.get('description'),
+    date_created: now,
+    date_updated: now
+  }).save()
+  .then(function(p) {
+    if (!p) { throw Boom.notFound(); }
+
+    project = p;
+    return publishedProject.publishedFiles().query({}).fetchAll();
+  })
+  .then(function(files) {
+    if(!files || !files.length) { throw Boom.badImplementation(); }
```

Comment:
I'm pretty sure a `fetchAll()` always returns something, so this should just be `if (!files.length)`